Minesweeper.java

```diff
@@ -81,24 +105,57 @@ private void run() {
             setVisible(true);
         }
 
+        public void revealEmptyCells(int row, int col) {
+            // Check bounds to avoid array out-of-bounds exceptions
+            if (row < 0 || col < 0 || row >= Game.cells.length || col >= Game.cells[0].length) {
+                return;
+            }
+            
+            Cell cell = Game.cells[row][col];
+            
+            // Check if the cell is already revealed or contains a mine
+            if (cell.isRevealed || cell.isBomb) {
+                return;
+            }
+            
+            // Reveal the cell
+            Game.cells[row][col].reveal();
+            
+
+            // If this cell has no adjacent mines, recursively reveal its neighbors
+            if (cell.getNeighboringBombs(Game.cells[row][col]) == 0) {
+                // Define relative positions of neighboring cells
+                int[] dr = {-1, -1, -1, 0, 1, 1, 1, 0};
```

Comment:
that's just the absolute values for the 9x9 grid encapsulating the cell, for me it makes more sense since we just a single for loop to check things

Reply:
you are using a single loop but that doesn't make it any faster because the logic is the same. the double for loop for neighboring cells will always only go through 8 (technically 9 but one is always skipped) cells so it is not quadratic

Reply:
yeah true, since it is the same, i dont think it needs changing

Reply:
the readability problem is still a thing, what does the `d` in `dr` and `dc` stand for?

Reply:
latest commit addresses this, have a look